jupyter_client/multikernelmanager.py

```diff
@@ -56,6 +56,8 @@ class MultiKernelManager(LoggingConfigurable):
         """
     )
 
+    cache_ports = Bool(True)
```

Comment:
The trait should be on the `KernelManager` since it's the `KernelManager` that is aware of the kernel it's managing - including its locality (and whether ports should be cached).